### PR TITLE
Make tag case-sensitivity a parameter in database service methods

### PR DESCRIPTION
**Pull request by demiankatz**
Historically, tag case sensitivity has worked "magically" based on configuration. In order to better separate database services from application details, I think it makes more sense for this to be a parameter to database methods, and pass it down from a central location in the TagsService. Admittedly this has some disadvantages (verbosity, possibility of inconsistency bugs), but I think it enables better separation of concerns.

### CODE REVIEW

File: module/VuFind/src/VuFind/Controller/MyResearchController.php [hunk 1]
```diff
@@ -1277,9 +1272,9 @@ public function editlistAction()
 
         $listTags = null;
         if ($this->listTagsEnabled() && !$newList) {
-            $tagService = $this->getDbService(TagServiceInterface::class);
+            $tagsService = $this->getDbService(TagsService::class);
```
Comment (EreMaijala) [inline module/VuFind/src/VuFind/Controller/MyResearchController.php]: Shouldn't this be `$this->serviceLocator->get(TagsService::class)`?

File: module/VuFind/src/VuFind/Db/Row/User.php [hunk 2]
```diff
@@ -296,8 +296,7 @@ public function checkEmailVerified()
      */
     public function getTags($resourceId = null, $listId = null, $source = null)
     {
-        return $this->getDbService(TagServiceInterface::class)
-            ->getUserTagsFromFavorites($this, $resourceId, $listId, $source);
+        return $this->getDbTable('Tags')->getListTagsForUser($this->getId(), $resourceId, $listId, $source);
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/Db/Row/User.php:299]: Have to roll these back to take advantage of the legacy null case sensitivity value.

**Review by EreMaijala — CHANGES_REQUESTED** [review]

File: module/VuFind/src/VuFind/Controller/MyResearchController.php (hunk 1 above)
Comment (demiankatz) [reply module/VuFind/src/VuFind/Controller/MyResearchController.php]: Are you reviewing an older revision? I already fixed that one a while ago...

Comment (EreMaijala) [reply module/VuFind/src/VuFind/Controller/MyResearchController.php]: Apparently yes, though not sure how that happened. Sorry!

**Review by EreMaijala — APPROVED** [review]

File: module/VuFind/src/VuFind/Controller/MyResearchController.php (hunk 1 above)
Comment (demiankatz) [reply module/VuFind/src/VuFind/Controller/MyResearchController.php]: No problem; sorry for wasting your time by making such a silly mistake in the first place. ;-)

**Pull request merged**